packages/zowe-explorer/src/trees/job/JobFSProvider.ts

```diff
@@ -262,11 +271,25 @@ export class JobFSProvider extends BaseProvider implements vscode.FileSystemProv
         }
 
         this._fireSoon({ type: vscode.FileChangeType.Changed, uri });
-        if (query.has("startLine") && !query.has("endLine")) {
+
+        if (recordsToFetch === 1) {
+            if (this.allLines.length === 0) {
+                const content = bufBuilder.read()?.toString("utf-8") ?? "";
+                this.allLines = content.split(/\r?\n/).filter((line) => line.trim() !== "");
+                this.currentLine = 0;
+            }
+            if (this.currentLine < this.allLines.length) {
+                const line = this.allLines[this.currentLine];
+
+                spoolEntry.data = Buffer.concat([spoolEntry.data, Buffer.from(line + "\n", "utf-8")]);
+                this.currentLine += 1;
+            }
```

Comment:
I think we'll need to avoid putting `allLines` and `currentLine` directly on the jobs FS provider as this would cause undefined behavior with 2 spool files open & pagination enabled.

Question for my own clarification: is there a reason why we need different logic here when the records to fetch is 1?

It seems like if one record was pulled using record range, the same method of `Buffer.concat(...)` on line 288 would work to add the new line to the existing contents. But maybe I'm missing something 😋